tools/openocd/defs.bzl

```diff
@@ -92,20 +94,66 @@ def _openocd_debug_server_impl(ctx):
     if ctx.attr.transport:
         transport_string = "-c \"transport select %s\"" % (ctx.attr.transport)
     script_template = """
-{openocd} {interface_config_string} {transport_string} -c "adapter_khz {programmer_frequency}" {chip_config_string}
+{openocd} {interface_config_string} {transport_string} -c "adapter speed {programmer_frequency}" {chip_config_string}
 
 """
 
-    script = ctx.actions.declare_file("%s.sh" % ctx.label.name)
-    script_content = script_template.format(
-        openocd = ctx.file._openocd.short_path,
-        interface_config_string = interface_config_string,
-        chip_config_string = chip_config_string,
-        programmer_frequency = ctx.attr.programmer_frequency,
-        transport_string = transport_string,
-    )
+    script_content = ""
+    if is_windows:
+        # A bit of a hacky way to do this... The issue is Windows does not support
+        # symlinks which does not work well for runfiles in particular. To enable them
+        # the user must use Developer Mode, also not ideal. Instead:
+        # 1) Use the 'symlink' as a generic path to the root of openocd
```

Comment:
Just to consolidate my understanding here does this use symlinks and hence require developer mode. Or is this simulating symlinks (with inverted quotes).

Reply:
This does *not* use symlinks or developer mode.

The best that I can tell, when repository_ctx.symlink() is called, it simply copies the file to the new location. So this line creates a copy of the exe at the root of the openocd directory, but without the supporting files next to it

https://github.com/silvergasp/bazel-embedded/blob/34004ad4b06efa17450b64d21fb46d3a8a9f78ef/tools/openocd/openocd_repository.bzl#L48

This file is also the only entry in the MANIFEST file without making other changes, but its good enough to give a path to the actual files. So the batch script finds the entry that starts with com_openocd, then strips the fake symlink name and replaces is with /bin/openocd.exe

Reply:
I'll disable developer mode next time I'm booted on windows, and give this a proper look.